src/window.rs

```diff
@@ -438,9 +440,31 @@ pub fn ui_loop() {
                     keycode: received_keycode,
                     ..
                 } => {
+                    if received_keycode.is_some() {
+                        keyboard_inputs.push((keycode, None));
+                    }
+
                     keycode = received_keycode;
+
+                    if keytext.is_some() {
+                        keyboard_inputs.push((keycode, keytext));
+                        keycode = None;
+                        keytext = None;
+                    }
+                }
+                Event::TextInput { text, .. } => {
+                    if keytext.is_some() {
```

Comment:
This is where I'm afraid the double happens.

Reply:
id be interested in seeing how it works on non windows platforms. It doesn't double up on windows

Reply:
Grabbing now.

Reply:
There isn't any double up.